pkg/identityserver/gormstore/notification.go

```diff
@@ -0,0 +1,132 @@
+// Copyright © 2022 The Things Network Foundation, The Things Industries B.V.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package store
+
+import (
+	"time"
+
+	"github.com/gogo/protobuf/types"
+	"github.com/jinzhu/gorm/dialects/postgres"
+	"github.com/lib/pq"
+	"go.thethings.network/lorawan-stack/v3/pkg/jsonpb"
+	"go.thethings.network/lorawan-stack/v3/pkg/ttnpb"
+)
+
+// Notification model.
+type Notification struct {
+	Model
+
+	EntityID   string `gorm:"type:UUID;index:notification_entity_index;not null"`
+	EntityType string `gorm:"type:VARCHAR(32);index:notification_entity_index;not null"`
+
+	NotificationType string `gorm:"not null"`
+
+	Data postgres.Jsonb `gorm:"type:JSONB"`
+
+	Sender   *User
+	SenderID *string `gorm:"type:UUID;index:notification_sender_index"`
+
+	Receivers pq.Int32Array `gorm:"type:INT ARRAY"`
```

Comment:
Do we need this?

In `ListNotifications` we are still performing join with `NotificationReceiver` so we are not really saving any query time by including this.

Reply:
This contains the list of `ttnpb.NotificationReceiver`, not the IDs of actual receivers. So it indicates that this notification was sent to (one or more of): administrative contact, technical contact, collaborators.